### PR TITLE
Blood 🩸 

### PR DESCRIPTION
**Pull request by madrury**
Extends the visual effect of blood (spilled during melee combat) to grass and water. For now, this is a visual only change with no gameplay consequences, though it opens up some design space.

Screenshot 2024-02-18 at 6 44 28 PM

## New Components
We add these new components:

  - `Bloodied`: Tags entities that are bloodied.
  - `UseFgColorMapWhenBloodied`: If present, swap to a new fg colormap when bloodied.
  - `UseBgColorMapWhenBloodied`: If present, swap to a new bg colormap when bloodied.

## Systems
`ColorSynchronizationSystem` runs every sub-turn, and swaps the colormap when a tile becomes bloodied. For now, this is a one way door: there's no process to swap the colormap back to the original. A tile is bloodied or it is not, and once bloodied it stays bloodied.

### CODE REVIEW

File: rogue/src/entity_spawners/mod.rs [hunk 1]
```diff
@@ -117,17 +117,18 @@ fn spawn_random_item(ecs: &mut World, x: i32, y: i32, depth: i32) {
...
             .insert(ItemType::BlinkScroll, depth)
             .insert(ItemType::FireblastScroll, depth)
-            // .insert(ItemType::FireballScroll, depth)
             .insert(ItemType::IceblastScroll, depth)
-            // .insert(ItemType::IcespikeScroll, depth)
             .insert(ItemType::None, 75)
+            // These are avalable at blessing statues.
+            // .insert(ItemType::MagicMissileScroll, depth)
+            // .insert(ItemType::FireballScroll, depth)
+            // .insert(ItemType::IcespikeScroll, depth)
```
Comment (axiomabsolute) [inline rogue/src/entity_spawners/mod.rs:131]: Were these taken out intentionally?

File: rogue/src/main.rs [hunk 2]
```diff
@@ -300,11 +301,16 @@ impl State {
...
+        }
     }
 
-    fn run_map_indexing_system(&mut self) {
-        let mut mis = MapIndexingSystem{};
-        mis.run_now(&self.ecs);
+    fn run_synchronization_systems(&mut self) {
+        let mut mss = MapSynchronizationSystem{};
+        mss.run_now(&self.ecs);
+        let mut css = ColorSynchronizationSystem{};
```
Comment (axiomabsolute) [inline rogue/src/main.rs:312]: CSS!? One step closer to the JavaScript frontend :trollface:

**Review by axiomabsolute — APPROVED** [review]
One question about whether the scroll-dropping changes were intentional, otherwise :+1:  Straightforward.

File: rogue/src/entity_spawners/mod.rs (hunk 1 above)
Comment (madrury) [reply rogue/src/entity_spawners/mod.rs:131]: Yah, I left the comment to hopefully let that be clear, but left them commented out for debugging ease.

File: rogue/src/main.rs (hunk 2 above)
Comment (madrury) [reply rogue/src/main.rs:312]: Omg, cursed.

**Pull request merged**